datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1831,8 +1832,13 @@ fn from_substrait_type(
                 Ok(DataType::Interval(IntervalUnit::YearMonth))
             }
             r#type::Kind::IntervalDay(_) => Ok(DataType::Interval(IntervalUnit::DayTime)),
```

Comment:
Yup see https://github.com/apache/datafusion/pull/12112#discussion_r1818961187. I think we should change it, maybe with some option so that user can decide if they want to throw or accept the precision loss. But will be a separate PR :)